_posts/2024-10-02-#30793.md

```diff
@@ -10,13 +10,53 @@ status: upcoming
 commit:
 ---
 
-_Notes and Questions to follow soon!_
+## Notes
 
-<!-- ## Notes -->
+- An orphan transaction is a transaction with missing inputs. The p2p code uses a [`TxOrphanage`](https://github.com/bitcoin/bitcoin/blob/842f7fdf786fcbbdf3df40522945813404f8a397/src/net_processing.cpp#L1079-L1081) to store orphan transactions, to be reconsidered later if/when its parent transaction(s) are known.
 
-<!-- ## Questions -->
+    - We have discussed `TxOrphanage` in previous review club meetings [n21527](/21527) and [n30000](/30000).
 
-<!-- 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach? -->
+- [PR #30793](https://github.com/bitcoin/bitcoin/pull/30793) adds a new RPC, `getorphantxs`, to
+  return the contents of the node's orphanage at that moment.
+
+    - Its format is similar to the `getrawmempool` RPC, which also returns information on all transactions in the mempool. Lower verbosity returns the txids,
+      and higher verbosity returns fields about each entry.
+
+    - Its purpose is similar to that of `getrawaddrman`. Most likely, developers will be the main users.
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
+
+1. What is an orphan transaction? At what point do transactions enter the orphanage (can you find the code)?
+
+1. What command can you run to get the list of available RPCs?
+
+1. What is the benefit of the first commit, which creates a `public OrphanTxBase` and extends that in a `protected struct OrphanTx`?
+
+1. What is the difference between `public`, `protected`, and `private`? When should you use each for
+   a class member/method?
+
+1. If an RPC has a non-string argument, does anything special need to be done to handle it?
+
+1. What specifically does it mean that the RPC is
+   ["hidden"](https://github.com/bitcoin-core-review-club/bitcoin/commit/8ec094959dc6afd46a709190d2deb58a50593fb7#diff-9c5b83de6dc84af277e352c88b9291aa44340a3c75f572a0b51661eb0a838de9R1131)? Why `hidden` and not `net`?
+
+1. Why can't we just directly access the orphanage from the RPC code? Why don't we just add a
+   PeerManager function that returns a reference to the `TxOrphanage`, which would be more extensible?
+
+1. What is the maximum size of the result from this RPC? Is there a limit to how many orphans are retained?
```

Comment:
Could add a hint here that aids the reader in later answering "able to tell how long a transaction has been in the orphanage".  Maybe:

```diff
- 1. What is the maximum size of the result from this RPC? Is there a limit to how many orphans are retained?
+ 1. What is the maximum size of the result from this RPC? Is there a limit to how many orphans are retained? Is there a limit to how much time orphans are retained?
```

Reply:
Sure, added